<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SetupDialog</class>
 <widget class="QDialog" name="SetupDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>752</width>
    <height>673</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string notr="true">Data Logger</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tab_widget">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab_setup">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <attribute name="title">
       <string>Setup</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <layout class="QGridLayout" name="gridLayout">
         <item row="2" column="0">
          <widget class="QGroupBox" name="group_debug_file">
           <property name="title">
            <string>Debug</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <layout class="QGridLayout" name="gridLayout_9">
              <item row="3" column="0">
               <widget class="QLabel" name="label_log_level">
                <property name="text">
                 <string>Log Level:</string>
                </property>
               </widget>
              </item>
              <item row="3" column="1">
               <widget class="QComboBox" name="combo_log_level"/>
              </item>
              <item row="0" column="1">
               <widget class="QComboBox" name="combo_debug_time_format">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                  <horstretch>1</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
               </widget>
              </item>
              <item row="0" column="0">
               <widget class="QLabel" name="label_4">
                <property name="text">
                 <string>Time Format:</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <layout class="QHBoxLayout" name="horizontalLayout_3">
                <item>
                 <widget class="QLineEdit" name="edit_log_file_name"/>
                </item>
                <item>
                 <widget class="QPushButton" name="btn_browse_log_file_name">
                  <property name="text">
                   <string>Browse</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="label_log_file_name">
                <property name="text">
                 <string>Log File:</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QCheckBox" name="check_debug_to_log_file">
                <property name="text">
                 <string>Write to Log File</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QGroupBox" name="group_general">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="title">
            <string>General</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <layout class="QGridLayout" name="gridLayout_2">
              <item row="2" column="1">
               <widget class="QCheckBox" name="check_authentication">
                <property name="text">
                 <string>Use Authentication</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QComboBox" name="combo_host">
                <property name="editable">
                 <bool>true</bool>
                </property>
                <property name="insertPolicy">
                 <enum>QComboBox::InsertAtTop</enum>
                </property>
                <property name="sizeAdjustPolicy">
                 <enum>QComboBox::AdjustToMinimumContentsLength</enum>
                </property>
               </widget>
              </item>
              <item row="0" column="0">
               <widget class="QLabel" name="label">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Host:</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="label_2">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Port:</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QSpinBox" name="spin_port">
                <property name="minimum">
                 <number>1</number>
                </property>
                <property name="maximum">
                 <number>65535</number>
                </property>
                <property name="value">
                 <number>4223</number>
                </property>
               </widget>
              </item>
              <item row="4" column="1">
               <widget class="QCheckBox" name="check_secret_show">
                <property name="text">
                 <string>Show Characters</string>
                </property>
               </widget>
              </item>
              <item row="3" column="0">
               <widget class="QLabel" name="label_secret">
                <property name="text">
                 <string>Secret:</string>
                </property>
               </widget>
              </item>
              <item row="3" column="1">
               <widget class="QLineEdit" name="edit_secret">
                <property name="toolTip">
                 <string>Warning: Secret is stored in plain text without any protection!</string>
                </property>
                <property name="maxLength">
                 <number>64</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QGroupBox" name="group_data_file">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="title">
            <string>Data</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <layout class="QGridLayout" name="gridLayout_3">
              <item row="0" column="0">
               <widget class="QLabel" name="label_7">
                <property name="text">
                 <string>Time Format:</string>
                </property>
               </widget>
              </item>
              <item row="4" column="0">
               <widget class="QLabel" name="label_csv_file_name">
                <property name="text">
                 <string>CSV File:</string>
                </property>
               </widget>
              </item>
              <item row="4" column="1">
               <layout class="QHBoxLayout" name="horizontalLayout_2">
                <item>
                 <widget class="QLineEdit" name="edit_csv_file_name"/>
                </item>
                <item>
                 <widget class="QPushButton" name="btn_browse_csv_file_name">
                  <property name="focusPolicy">
                   <enum>Qt::TabFocus</enum>
                  </property>
                  <property name="text">
                   <string>Browse</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item row="0" column="1">
               <widget class="QComboBox" name="combo_data_time_format">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                  <horstretch>1</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="maxVisibleItems">
                 <number>15</number>
                </property>
               </widget>
              </item>
              <item row="3" column="1">
               <widget class="QCheckBox" name="check_data_to_csv_file">
                <property name="text">
                 <string>Write to CSV File</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QLabel" name="label_data_time_format_strftime_help">
                <property name="text">
                 <string>See the Python &lt;a href=&quot;https://docs.python.org/2/library/datetime.html#strftime-and-strptime-behavior&quot;&gt;strftime&lt;/a&gt; documentation for details about the format syntax.</string>
                </property>
                <property name="openExternalLinks">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QLineEdit" name="edit_data_time_format_strftime">
                <property name="text">
                 <string>%Y%m%d_%H%M%S</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_devices">
      <attribute name="title">
       <string>Devices</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_8">
       <item row="0" column="2">
        <widget class="QPushButton" name="btn_remove_device">
         <property name="text">
          <string>Remove Device</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QPushButton" name="btn_add_device">
         <property name="text">
          <string>Add Device</string>
         </property>
        </widget>
       </item>
       <item row="0" column="4">
        <spacer name="horizontalSpacer_7">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="5">
        <widget class="QPushButton" name="btn_remove_all_devices">
         <property name="text">
          <string>Remove All Devices</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0" colspan="6">
        <widget class="QTreeView" name="tree_devices">
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::ExtendedSelection</enum>
         </property>
         <attribute name="headerStretchLastSection">
          <bool>true</bool>
         </attribute>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_data">
      <attribute name="title">
       <string>Data</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="0" column="0" colspan="2">
          <widget class="QLabel" name="label_3">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>1</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Shows the latest 1000 data entries</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QPushButton" name="btn_clear_data">
           <property name="text">
            <string>Clear</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0" colspan="3">
          <widget class="QTableView" name="table_data">
           <property name="editTriggers">
            <set>QAbstractItemView::NoEditTriggers</set>
           </property>
           <property name="alternatingRowColors">
            <bool>true</bool>
           </property>
           <attribute name="horizontalHeaderStretchLastSection">
            <bool>true</bool>
           </attribute>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_debug">
      <attribute name="title">
       <string>Debug</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_7">
       <item>
        <layout class="QGridLayout" name="gridLayout_5">
         <item row="1" column="0" colspan="4">
          <widget class="QTextBrowser" name="text_debug">
           <property name="lineWrapMode">
            <enum>QTextEdit::NoWrap</enum>
           </property>
          </widget>
         </item>
         <item row="0" column="3">
          <widget class="QPushButton" name="btn_clear_debug">
           <property name="focusPolicy">
            <enum>Qt::TabFocus</enum>
           </property>
           <property name="text">
            <string>Clear</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QCheckBox" name="checkbox_debug_auto_scroll">
           <property name="text">
            <string>Auto Scroll</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_5">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>1</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Shows the latest 1000 debug messages</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QComboBox" name="combo_debug_level">
           <property name="currentIndex">
            <number>-1</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="btn_start_logging">
       <property name="focusPolicy">
        <enum>Qt::TabFocus</enum>
       </property>
       <property name="text">
        <string>Start Logging</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_5">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="btn_load_config">
       <property name="focusPolicy">
        <enum>Qt::TabFocus</enum>
       </property>
       <property name="text">
        <string>Load Config</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_save_config">
       <property name="focusPolicy">
        <enum>Qt::StrongFocus</enum>
       </property>
       <property name="text">
        <string>Save Config</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_close">
       <property name="text">
        <string>Close</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>tab_widget</tabstop>
  <tabstop>combo_host</tabstop>
  <tabstop>spin_port</tabstop>
  <tabstop>check_authentication</tabstop>
  <tabstop>edit_secret</tabstop>
  <tabstop>check_secret_show</tabstop>
  <tabstop>combo_data_time_format</tabstop>
  <tabstop>edit_data_time_format_strftime</tabstop>
  <tabstop>check_data_to_csv_file</tabstop>
  <tabstop>edit_csv_file_name</tabstop>
  <tabstop>btn_browse_csv_file_name</tabstop>
  <tabstop>combo_debug_time_format</tabstop>
  <tabstop>check_debug_to_log_file</tabstop>
  <tabstop>edit_log_file_name</tabstop>
  <tabstop>btn_browse_log_file_name</tabstop>
  <tabstop>combo_log_level</tabstop>
  <tabstop>btn_start_logging</tabstop>
  <tabstop>btn_load_config</tabstop>
  <tabstop>btn_save_config</tabstop>
  <tabstop>btn_add_device</tabstop>
  <tabstop>btn_remove_device</tabstop>
  <tabstop>btn_remove_all_devices</tabstop>
  <tabstop>tree_devices</tabstop>
  <tabstop>table_data</tabstop>
  <tabstop>btn_clear_data</tabstop>
  <tabstop>text_debug</tabstop>
  <tabstop>checkbox_debug_auto_scroll</tabstop>
  <tabstop>combo_debug_level</tabstop>
  <tabstop>btn_clear_debug</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
